Backport aio circleci patch to v5

petebacondarwin opened the pull request:
Assigning to @IgorMinar  as he is probably best placed to get this approved and merged into 5.2.x.

petebacondarwin committed:
ci(docs-infra): update upload-server to run on node.js v10

petebacondarwin committed:
ci(docs-infra): move AIO preview deployment to CircleCI

Now instead of pushing the AIO build artifacts to the preview server
from inside a Travis job, the artifacts are built and hosted on the
CircleCI infrastructure. The preview server will then pull these
down after being triggered by a CircleCI build webhook.

# Conflicts:
#	.circleci/config.yml

petebacondarwin committed:
docs(docs-infra): update the preview server documentation

# Conflicts:
#	aio/aio-builds-setup/docs/vm-setup--start-docker-container.md

petebacondarwin committed:
ci(docs-infra): add helper scripts for running TDD in Docker

petebacondarwin committed:
ci(docs-infra): move the payload-size check to the test job

petebacondarwin committed:
ci(docs-infra): factor out the aio-builds-setup environment variables

petebacondarwin committed:
ci(docs-infra): change AIO preview server stuff to pull builds from CircleCI

Previously, Travis pushed the build artitfacts to the preview server.
This required us to use JWT to secure the POST request from Travis, to
ensure we couldn't receive malicious builds.

JWT has been deprecated and we are moving our builds to CircleCI.

This commit rewrites the TypeScript part of the preview server that
handles converting build artifact into hosted previews of the docs.

petebacondarwin committed:
ci(docs-infra): improve preview-server logging

petebacondarwin committed:
ci(docs-infra): add explicit return types to methods

petebacondarwin committed:
ci(docs-infra): rename 'upload-server' to 'preview-server'

The server no longer has files uploaded to it. Instead it is more
accurate to refer to it as dealing with "previews" of PRs.

petebacondarwin committed:
docs(docs-infra): the `build.sh` script was renamed to `create-image.sh`

mary-poppins commented:
You can preview 6d92984 at https://pr25636-6d92984.ngbuilds.io/.

IgorMinar reviewed (approved):
is it expected that the CI is not happy? can you try to green it please?

petebacondarwin commented:
Will do...

petebacondarwin committed:
ci: add `docs-infra` as a commit message scope

petebacondarwin committed:
fix(compiler): remove bogus lines

These seem to have been added by mistake in
408d8cb735085bec4dcfab17bcf17b66368b98f2

mary-poppins commented:
You can preview d913ce9 at https://pr25636-d913ce9.ngbuilds.io/.

mary-poppins commented:
You can preview ea582d9 at https://pr25636-ea582d9.ngbuilds.io/.
